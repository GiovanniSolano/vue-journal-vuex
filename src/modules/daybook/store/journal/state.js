 export default () => ({
    isLoading: true,
    entries: [
       {
          id: new Date().getTime(),
          date: new Date().toDateString(),
          text: 'Anim commodo non magna est culpa sit nisi laborum. Incididunt ea reprehenderit esse consequat et cillum ad ea dolore nulla. Et velit ut cillum ullamco fugiat exercitation eu elit ad. Non aliquip nostrud eu ea tempor anim consequat. Labore ea eiusmod do nostrud ea aliquip dolore sint dolor officia ipsum fugiat et do.',
          picture: null
       },
       {
          id: new Date().getTime() + 12,
          date: new Date().toDateString(),
          text: 'esse consequat et cillum ad ea dolore nulla. Et velit ut cillum ullamco fugiat exercitation eu elit ad. Non aliquip nostrud eu ea tempor anim consequat. Labore ea eiusmod do nostrud ea aliquip dolore sint dolor officia ipsum fugiat et do.',
          picture: null
       },
       {
          id: new Date().getTime() + 120,
          date: new Date().toDateString(),
          text: 'a. Et velit ut cillum ullamco fugiat exercitation eu elit ad. Non aliquip nostrud eu ea tempor anim consequat. Labore ea eiusmod do nostrud ea aliquip dolore sint dolor officia ipsum fugiat et do.',
          picture: null
       },
    ]
 })