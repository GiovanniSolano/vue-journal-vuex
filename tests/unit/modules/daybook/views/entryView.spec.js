import { shallowMount } from '@vue/test-utils'
import { createStore } from 'vuex'

import Swal from "sweetalert2";

import journal from '@/modules/daybook/store/journal'
import { journalState } from "../../../mock-data/test-journal-state"

import EntryView from "@/modules/daybook/views/EntryView.vue";


const createVuexStore = ( initialState ) => createStore({
    modules: {
        journal: {
            ...journal,
            state: { ...initialState }
        }
    }
})

jest.mock('sweetalert2', () => ({
    fire: jest.fn(),
    showLoading: jest.fn(),
    close: jest.fn()
}))



describe('Pruebas en el EntryView', () => {

    const store = createVuexStore(journalState)
    store.dispatch = jest.fn()
    
    const mockRouter = {
        push: jest.fn()
    }

    let wrapper;

    beforeEach(() => {

        jest.clearAllMocks()
        wrapper = shallowMount(EntryView, {
            props: {
                id: '-MqlURWIow14EuuQuvnI'
            },
            global: {
                mocks: {
                    $router: mockRouter
                },
                plugins: [store]
            }
        })
    })

    test('debe de sacar al usuario porque el id no existe', () => {
        

        const wrapper = shallowMount(EntryView, {
            props: {
                id: 'Este id no existe en el store'
            },
            global: {
                mocks: {
                    $router: mockRouter
                },
                plugins: [store]
            }
        })

        expect(mockRouter.push).toHaveBeenCalledWith({name: 'no-entry'})

    })

    test('debe de mostrar la entrada correctamente', () => {
        
        expect(wrapper.html()).toMatchSnapshot()
        expect(mockRouter.push).not.toHaveBeenCalledWith()

    })

    test('debe de borrar la entrada y salir', (done) => {
        
        Swal.fire.mockReturnValueOnce(Promise.resolve({ isConfirmed: true }))


        wrapper.find('.btn-danger').trigger('click')

        expect(Swal.fire).toHaveBeenLastCalledWith(
            {
                title: 'Borrar registro?',
                text: 'Una vez borrado, no se puede recuperar',
                showDenyButton: true,
                confirmButtonText: 'Si, estoy seguro'
              }
        )

        setTimeout(() => {
            expect(store.dispatch).toHaveBeenCalledWith('journal/deleteEntry', '-MqlURWIow14EuuQuvnI')
            expect(mockRouter.push).toHaveBeenCalled()
            done()
        }, 1)

    })

})
