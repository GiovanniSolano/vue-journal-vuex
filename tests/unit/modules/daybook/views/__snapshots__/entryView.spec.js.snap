// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el EntryView debe de mostrar la entrada correctamente 1`] = `
<div class="entry-title d-flex justify-content-between p-2">
  <div><span class="text-success fs-3 fw-bold">12</span><span class="mx-1 fs-3">Diciembre</span><span class="mx-2 fs-4 fw-light">2021, Domingo</span></div><input accept="image/png, image/jpeg" type="file" style="display: none;">
  <div><button class="btn btn-danger mx-2"> Borrar <i class="fa fa-trash-alt"></i></button><button class="btn btn-primary"> Subir Foto <i class="fa fa-upload"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column px-3 h-75"><textarea placeholder="Que sucedio hoy?"></textarea></div>
<fab-stub icon="fa fa-save"></fab-stub>
<!--v-if-->
<!--v-if-->
`;
